homeassistant/components/tado/climate.py

```diff
@@ -435,11 +474,20 @@ def set_swing_mode(self, swing_mode):
     def _async_update_zone_data(self):
         """Load tado data into zone."""
         self._tado_zone_data = self._tado.data["zone"][self.zone_id]
+        self._update_offset()
         self._current_tado_fan_speed = self._tado_zone_data.current_fan_speed
         self._current_tado_hvac_mode = self._tado_zone_data.current_hvac_mode
         self._current_tado_hvac_action = self._tado_zone_data.current_hvac_action
         self._current_tado_swing_mode = self._tado_zone_data.current_swing_mode
 
+    def _update_offset(self):
+        # Assign offset values to mapped attributes
+        for key in TADO_TO_HA_OFFSET_MAP:
+            if key in self._tado.data["device"][self._device_id][TEMP_OFFSET]:
+                self._tado_zone_temp_offset[
+                    TADO_TO_HA_OFFSET_MAP[key]
+                ] = self._tado.data["device"][self._device_id][TEMP_OFFSET][key]
```

Comment:
```suggestion
        for offset_key, attr in TADO_TO_HA_OFFSET_MAP.items():
            if offset_key in self._tado.data["device"][self._device_id][TEMP_OFFSET]:
                self._tado_zone_temp_offset[
                    attr
                ] = self._tado.data["device"][self._device_id][TEMP_OFFSET][offset_key]
```

Reply:
Ahh yeah nice.. need to check the device id exists too, for the tests, as I think they are failing when no mock response exists so the entity ends up not existing

So will update with your suggestion